src/main/java/com/google/sps/servlets/AuthenticationServlet.java

```diff
@@ -0,0 +1,115 @@
+
+  /**
+   * Checks if the user is logged in with UserService and prints response in json.
+   *
+   * @param request  provides request information from the HTTP servlet
+   * @param response response object where servlet will write information to
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json;");
+    UserService userService = UserServiceFactory.getUserService();
+    Map<String, String> authResponse = new HashMap<>();
+
+    try {
+      if (userService.isUserLoggedIn()) {
+        // If logged in get email and create link to logout
+        String userEmail = userService.getCurrentUser().getEmail();
+        String logoutUrl = userService.createLogoutURL("/");
+
+        authResponse.put("email", userEmail);
+        authResponse.put("logoutUrl", logoutUrl);
+
+        // Get User Info and add it to the response
+        authResponse.putAll(getUserInfo(userService.getCurrentUser().getEmail()));
+      } else {
+        // If logged out get login link
+        String loginUrl = userService.createLoginURL("/");
+        authResponse.put("loginUrl", loginUrl);
+      }
+    } catch (Exception e) {
+      logger.atSevere().log("There was an error: %s", e);
+      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+    }
+
+    String json = UtilityClass.convertToJson(authResponse);
+
+    // Write response to /auth
+    response.getWriter().println(json);
+  }
+
+  /**
+   * Returns the preferences of the user with email, or null if the user has not set their
+   * preferences.
+   *
+   * @param email the email of logged in user to retrieve linked userInfo from datastore
+   * @return a map of userInfo linked to the user email
+   */
+  private Map<String, String> getUserInfo(String email) {
+    Map<String, String> userInfoResponse = new HashMap<>();
+    try {
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      Query query =
+          new Query("UserInfo")
+              .setFilter(new Query.FilterPredicate("email", Query.FilterOperator.EQUAL, email));
+      PreparedQuery results = datastore.prepare(query);
+      Entity entity = results.asSingleEntity();
+      if (entity == null) {
+        userInfoResponse.put("updateInfoRequired", "true");
+        return userInfoResponse;
+      }
+      userInfoResponse.put("name", (String) entity.getProperty("name"));
+      userInfoResponse.put("font", (String) entity.getProperty("font"));
+      userInfoResponse.put("font_size", (String) entity.getProperty("font_size"));
+      userInfoResponse.put("bg_color", (String) entity.getProperty("bg_color"));
+      userInfoResponse.put("text_color", (String) entity.getProperty("text_color"));
+      return userInfoResponse;
+    } catch (Exception e) {
+      userInfoResponse.put("errorMsg", "Something went wrong. Please try again later.");
```

Comment:
Did you mean sendError() ?

Reply:
In this specific case I think it's better to provide an error in the JSON response as it allows users to still be authenticated even if there is an issue retrieving their preferences for colours, fonts etc.